build_tools/azure/debian_atlas_32bit_lock.txt

```diff
@@ -4,11 +4,14 @@
 #
 #    pip-compile --output-file=build_tools/azure/debian_atlas_32bit_lock.txt build_tools/azure/debian_atlas_32bit_requirements.txt
 #
+--index-url https://d1yxz45j0ypngg.cloudfront.net/
+--extra-index-url https://pypi.org/simple
```

Comment:
@ogrisel This is strange. I do not think the nogil index should appear in the 32bit lock file?

Reply:
I don't really understand, because in principle the script creates environment to call `pip-compile` from:
https://github.com/scikit-learn/scikit-learn/blob/996c7e2895db313dc7a5215473536f2b104121d0/build_tools/update_environments_and_lock_files.py#L485-L510

Reply:
I don't understand either, I cannot find where this custom `index-url` comes from.  I did not use my `nogil-venv` when running this command.

I also tried to manually activate the conda env used to run `pip-compile` for the debian CI entry and there is not custom pip config in it:


```
(base) $ conda activate pip-tools-python3.9.2
(pip-tools-python3.9.2) $ pip config debug
env_var:
env:
global:
  /etc/xdg/xdg-ubuntu-xorg/pip/pip.conf, exists: False
  /etc/xdg/pip/pip.conf, exists: False
  /etc/pip.conf, exists: False
site:
  /home/ogrisel/mambaforge/envs/dev/envs/pip-tools-python3.9.2/pip.conf, exists: False
user:
  /home/ogrisel/.pip/pip.conf, exists: False
  /home/ogrisel/.config/pip/pip.conf, exists: False
```

Reply:
I printed the command executed by the script and it is:

```
/home/ogrisel/mambaforge/envs/dev/envs/pip-tools-python3.9.2/bin/pip-compile --upgrade build_tools/azure/debian_atlas_32bit_requirements.txt -o build_tools/azure/debian_atlas_32bit_lock.txt
```

If I delete the `build_tools/azure/debian_atlas_32bit_lock.txt` and regenerate it by running the command manually, I still get the custom `--index-url` ...

Reply:
@thomasjpfan feel free to re-run the command and git commit / push to this PR to fix it because I don't know how to fix my local setup.